torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1457,7 +1458,7 @@ def _get_orig_params(
         try:
             while True:
                 param = next(param_gen)
-                if param not in ignored_params and not isinstance(param, FlatParameter):
+                if param not in ignored_params and not _is_flattened(param):
```

Comment:
I'm wondering why we need `_is_flattened`, can't we just check type(param) == FlatPram?

Reply:
This is not important for this PR, but when we expose original parameters, we want to ignore the original parameters that have already been flattened.

`not _is_flattened(param)` correctly captures the logical intention of this method `_get_orig_params()`. We only want this method to return the parameters that have not already been considered by FSDP previously (in some previous FSDP constructor). We previously used `isisnstance(param, FlatParameter)` to determine this because we had an invariant that after an FSDP constructor, all original parameters that have been flattened are de-registered as parameters and only their corresponding `FlatParameter`s are registered.